### DISCUSSION TITLE
Am I right that this specific ESP32 variant has not on-board antenna?

### DISCUSSION THREAD

**Discussion by Jibun-no-Kage** [micropython/micropython#10502] [General]
WARNING... A Hardware Question...

I have a use case where I have an existing ESP8266 that is inside a PC case, the ESP8266 drives a mofset so I can trip the power switch lead on the PC motherboard.  This means I can remote power on or even remote reset the PC as needed.

Given the ESP module is enclosed by steel, its WiFi signal strength is not great.  So I was thinking about soldering an external antenna to the existing module, cutting the on-board antenna leads, etc.  There are some nice videos and such no how to do this.  But I figured, heck there are ESP32 modules that have external IPEX ports, so maybe use that.

So here is the question... Am I right that this specific ESP32 variant that has external antenna IPEX port built in, has no on-board antenna?  So I don't have to switch a zero 0hm resister or bridge any contacts to disable the on-board antenna?  Looking at the picture I don't see an obvious place to disable the on-board antenna.

![image](https://user-images.githubusercontent.com/96628648/212525055-0ec1c89a-a1e2-49f1-81da-e7dd381cf240.png)

All the ESP8266 and ESP32 modules I have used in the past, the on-board antenna existence was obvious, but with above, I don't see it.  Or is it hidden under the rather large black rectangular area on the module?  And, of course, you could use a relay versus a mofset, but I find the mofset just easier to deal with.  But I digress.

**Comment by davefes** [comment]
Link doesn't work for me, one has to do a copy/paste.  That board doesn't have an on-board antenna.

**Comment by Jibun-no-Kage** [comment]
Oh, that link did not work... posting picture, replaced link above.  I just don't see any obvious place to by-pass any on-board antenna, usually there is a 0 ohm resistor or a 3 point area that is clearly now you disconnect the on-board and complete the connect for the IPEX port for the external antenna to be functional.

For example...

![image](https://user-images.githubusercontent.com/96628648/212525177-605006b6-a924-4118-a66e-68aff2ee8713.png)

**Comment by Jibun-no-Kage** [comment]
Of course, the vendor does not even reference the issue, and there is no way to contact them, thanks Amazon!  One customer said the Bluetooth range is only 1M, which REALLY at least to me, suggests something is odd.  Maybe they did not hook up any antenna?  If there was an on-board antenna functional by default I would think the Bluetooth would be better than 1M.

**Comment by robert-hh** [comment]
If your board matches the picture in the first post, then there is no on-board antenna, just the antenna connector. If you do not connect an antenna, it should still work being close to the module. The RF circuitry of ESPxx devices is quite sensitive. And the ones with on-board antenna will also work with reduced range inside a PC box. It is actually quite complicated to make a metal box EMC sealed. Every opening, every slit and every cable will let waves pass and can serve as secondary antenna.

**Comment by Jibun-no-Kage** [comment]
The range is only about 10 feet and the module was struggling badly, I changed modules, routers, etc., isolating one change at a time, the behavior was consistent.  So I removed the side wall of the PC case, and the issue disappeared.  It is an older HP workstation case, and it has really thick panels compared to typical cases now, no plastic or glass inserts, no fan ports, and the only venting is on the opposite side needed.  So I know the case is at issue.  I have used other cases, no significant problem but this case is very old school, if the ceiling crashed down, I would hide next to this case for sure, it is that robust. :)

I could not see any on board either.  I am tempted to purchase one of these just to test it.  Also considered using the side panel as the antenna as well, but decided that was too complicated to maintain and could not move the project module with ease when needed.  I am considering maybe testing a line wire antenna as well, but I have not take the time to work out the optimal wire length.  If I recall right for 2.4 GHz the antenna wire only has to be 23 or so mm.

**Reply by davefes** [reply]
On all my ESP32 projects I use these:
https://www.aliexpress.com/item/1899893948.html?spm=a2g0o.order_list.order_list_main.112.21ef1802J9yy1T
These work.  If used outside then you have to think about keeping water out of the antenna and the connector on the case.

As a transmitter designer for most of my life and probably spent most of that time dealing with affects of poor antenna loading I am surprised that the transmitter in the ESP32 chips can handle the abuse.  That board has no built-in antenna.  Fortunately, for you, the phase angle of the mismatch applied is not killing the RF part of the WiFi.

Why can you not mount an external antenna?

**Comment by Jibun-no-Kage** [comment]
Just for reference... I happen to find this reference...
https://circuitdigest.com/article/how-to-design-a-pcb-antenna-for-24ghz

Here is the page I read some time ago...
https://oscarliang.com/repair-2-4ghz-antenna-rx/

**Reply by davefes** [reply]
The 2nd article is worth a try, but is it still inside the metal box?

**Comment by Jibun-no-Kage** [comment]
I have been trying to avoid it, so it is not easy to break, in my world they are abused due to human mishap is all.  I was just intrigued by this board design that seemed, now believed, to have no on board antenna.  Before now, at least for me, every module I have seen with an IPEX port, needed that 0 ohm resistor change, and of course in the other common scenario, disabling the on board and adding an external antenna lead.

**Reply by davefes** [reply]
Yes, that seems a strange thing to do.  The other port is a very short stub not an antenna.  That second article is a rough attempt at making a coaxial-sleeve dipole.  If you take apart a common WiFi whip antenna there will be a large metal tube below the wire that forms the other half of the dipole.

Would it be possible to embed a patch antenna into the case.  I have not been impressed with PCB patch antennas on FR4 board but at least the RF would be on the outside of the box.

**Comment by davefes** [comment]
https://www.pasternack.com/t-calculator-microstrip-ant.aspx
but with probe feed:
https://www.researchgate.net/figure/Probe-Feed-For-Microstrip-Patch_fig2_312023108
could sit flat on the case.

**Reply by davefes** [reply]
https://www.antenna-theory.com/antennas/patches/antenna.php
then the link to feed methods

**Comment by Jibun-no-Kage** [comment]
Patch antenna?  Yes, that is possible too.  This is interesting discussion, I thought I had a simple question, but the design theory aspect is really interesting, at least to me. :)  A patch antenna would in effective be implicitly directional, no?

**Reply by davefes** [reply]
To a degree.  It has polarisation so how it is mounted will have some effect and yes most of the radiation will be off the front of the patch.  Polar plots will give you an idea.

However, getting the RF out of the box is a major step forward, gain and the direction of radiation will become minor considerations.  How far are you trying to establish WiFi over?

**Reply by davefes** [reply]
I have found a patch on FR4 board to have about the same gain as a common WiFi whip, when you are looking perpendicular to the patch surface.

**Reply by davefes** [reply]
That antenna-theory article doesn't tell you where to put the probe.  In a couple days I could give you the dimensions and probe placement that I tested.  Have you got an access to a VNA?  If not, anything attempt is going to be better than what you have at the moment.

**Reply by davefes** [reply]
How close could you mount the ESP32 from the patch?  In that 2nd article that you linked to, if the IPEX cable is teflon cable I could tell you how to connect that to the patch.  If the ESP32 is further away from the patch you are probably going to need connectors.  I could describe another option using RG316 telfon cable, no connectors required.

**Comment by Jibun-no-Kage** [comment]
Same room typical scenario so less than 20 feet, pretty much line of sight, other than when someone walks across the room, I have a 2.4 GHz AP in the room already.  The real issue is the case is an old school HP mid-tower case, that is much heavier gauge steel then the typical (cheap) PC case you see lately.  The case has some limited venting (wrong directional vector to router) and no plastic or glass windows or fan ports.  So although incomplete as signal cage, seems to be good enough that the internal antenna of the module struggles.

I tested multiple modules, placing the case various ways, when the venting was pointed to the router, a got a signal strength increase, so that pretty much explained the issue, so pulled the case side wall, which establish line of sight, and strength dramatically improved of course.  So problem identified.  Then I considered adding an external antenna, either in case, brute force signal, or external.

The external is fine as long as I can keep it protected or low profile, in my world... pets, humans that don't watch where they going, etc., poor antennas get trashed.  Just to make it easier, I am going to run an external antenna, out the back of the case via a pci/pcie slot plate, I can drill it.  I have some SMA connector leads, and either 2 or 3dBi antenna, just to do a simple test, left over from adding external antennas to a couple of old routers.  I also have some older 1M ESP01s left over from another project that I can modify, i.e. disable the on board antenna.

I did order one of the modules that triggered this thread, just for the fun of it.  I mean, you can always use another ESP module at some point, right?!  Looking at my parts box, I realized I have a couple of IPEX to SMA leads already.  LOL
